scan-external-contacts/config.yml

```diff
@@ -0,0 +1,144 @@
+    
+    Your OAuth client will need the following permissions (assigned through a role).
+    
+    To scan External Contacts, Relationships or Notes: `externalContacts:contact:view`
+    
+    To scan External Organizations: `externalContacts:externalOrganization:view`
+    
+
+    In this tutorial, we will use an OAuth client with Client Credentials Grant.
+    
+    Update this sample with your OAuth ClientID, ClientSecret, and your Genesys Cloud region.
+
+- title: Generate an OAuth client token
+  content: |
+    In order to call the External Contacts Scanning endpoint you will need to use your OAuth client to generate a token. See [Grant - Client Credentials](/api/rest/authorization/use-client-credentials.html) for more info
+
+- title: Sample Parameters
+  content: |
+    The sample code defines a set of parameters, specific to the sample, to make it easier to modify them and to adapt them to your needs.
+
+    `maxRecordsPerPage`: the External Contacts Scan API endpoints allow to retrieve from 10 (min) to 200 (max) records per page. Default value is 100.
+
+    `minAPIRequestInterval` and `minWaitTime`: the sample code forces a pause between 2 API requests. This is to make sure that if you run the script, you will not hit an API rate limit (300 requests/minute for the client token). The sample is currently configured for more or less 2 API requests/second (500ms interval).
+    
+    You can adapt the pause duration to your needs and current use of API in your organization.
+    
+    As the scan of External Contacts (for export/save) is likely not something that has to return immediately (no real-time needs), it shouldn't be a problem to make a pause, and even to execute your code via a nightly batch.
+
+    The sample also manages a maximum number of consecutive rate limit errors (the application will exit).
+
+    It is possible to export the External Contacts to a CSV file (and to configure the field separator) or to a raw JSON file (JSON Array of External Contacts).
+
+- title: Login
+  content: |
+    The main function first obtains an OAuth Client token.
+
+- title: Opening output file for export
+  content: |
+    A Stream (Write) is created to write the External Contacts to a file.
+
+    Depending on the type of export (CSV or raw JSON), a "header" string is written to the file.
+
+- title: Scan in loop
+  content: |
+    The main function will then initiate a loop, [scanning for External Contacts page per page](/api/rest/v2/externalcontacts/#get-api-v2-externalcontacts-scan-contacts), until all data is retrieved.
+
+- title: Process and Export retrieved data
+  content: |
+    If a page containing External Contacts is retrieved, they are processed and written to the output file (via the Stream) according to the type of export (CSV or raw JSON).
+
+    Note that an External Contacts Scan request can be successful while not bringing back any External Contacts (entities array attribute is empty).
```

Comment:
This seems a little hard to follow. Shouldn't the HTTP status code provide a response to what happened?